### ISSUE TITLE
Autocomplete component fails WCAG SC 4.1.3 Status Messages

### ISSUE THREAD

**Issue by clhenrick** [Esri/calcite-design-system#12433]
### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Summary

When a sighted user types a search term into a calcite-autocomplete component they are able to visually tell that results are populated in the listbox below the input. However, users of assistive tech such as screen readers have no way of knowing whether results have been populated for their search term or not, nor how many. 

This likely fails [WCAG SC 4.1.3 Status Messages (Level AA)](https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html) and could be remedied by using an ARIA live region that announces that results have been populated for a search term. 

There are some UX considerations around implementing this should be considered. It would likely be a poor UX to read aloud every item in the list box for a search term, so an alternative approach may be preferable for SR users such as reading aloud the number of results. [Karl Groves summarized this well in the A11Y Slack](https://web-a11y.slack.com/archives/C042TSFGN/p1697128103184759?thread_ts=1697125471.760949&cid=C042TSFGN):

> This is a far more complex interaction than most people realize.  The user needs to be able to type the string of text with no interruptions or interference. The screen reader reads back the characters that are typed by the user. That readback lets the user know if they’ve mistyped. However, as you’ve noted, the user needs to know that results exist, how many exist, and how to get to them. They also need to be able to pause and restart typing, including erasing prior characters, etc.  All of this makes autocomplete really hard to get right. Most apps that offer autocomplete do it badly and most AT handle autocompletes badly.

> On the Web, we can use Live Regions for that.  On iOS there’s an equivalent to Live Regions, which would be in UIKIt’s UIAccessibility.post()  which would allow you to post a notification. *With the Live Region it is mostly a case of pausing to determine if the user is still typing before making an announcement to avoid stepping on the user’s feedback from typing.*

### Actual Behavior

Typing a search term into a calcite-autocomplete component visually shows results but the content change is not noticeable to screen reader users. 

When navigating the search results using the up and down arrow keys nothing is read aloud either. This likely causes separate WCAG violations from SC 4.1.3.

### Expected Behavior

Typing a search term into a calcite-autocomplete component should both visually shows results and be conveyed to screen reader users to adhere to WCAG SC 4.1.3.

When navigating the autocomplete results in the listbox using the up and down arrow keys, their contents should be announced by the screen reader.

### Reproduction Sample

https://developers.arcgis.com/javascript/latest/storybook/map-components/iframe.html?singleStory=true&globals=&id=arcgis-search--demo

### Reproduction Steps

1. Go to the JS SDK arcgis-search component docs demo
2. Use a screen reader software such as VoiceOver with Safari when searching for an address
3. Type into the autocomplete component's input something like "555 main street"
4. Notice that nothing is announced to the screen reader that indicates search results are populated
5. Navigate the list box search results using the down arrow key and notice nothing is read aloud

### Reproduction Version

3.2.1

### Relevant Info

MacOS Sequoia 15.5 (24F74)
Safari Version 18.5 (20621.2.5.11.8)
VoiceOver

Note that I have not fully tested with NVDA or JAWS or VoiceOver on iOS or Talkback on Android.

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Instant Apps

**Comment by isaacbraun** [comment]
The `aria-live` seems to be working, although it reads out every option as they appear (the results for each query as letters are typed) even though the current (last) query is displaying different results. I assume this is not something we should implement, but rather on the browser/AT side of things.

When cycling through the results using up/down arrows, VoiceOver and JAWS do not read out each option while NVDA just repeats the current query. Does we need to implement something for this?

cc @geospatialem @DitwanP

**Comment by geospatialem** [comment]
> The `aria-live` seems to be working, although it reads out every option as they appear (the results for each query as letters are typed) even though the current (last) query is displaying different results. I assume this is not something we should implement, but rather on the browser/AT side of things.

Yes, this is expected, so assistive technology would be provided with the results that are visually provided to users. We could work to finesse this a bit in the future where we could provide the number of results first, but it seems like would be an enhancement request or something we could wait for someone to file. WDYT, @isaacbraun?

> When cycling through the results using up/down arrows, VoiceOver and JAWS do not read out each option while NVDA just repeats the current query. Does we need to implement something for this?

Great catch with the individual `autocomplete-item`. 🎣 Yes, this is a different bug, but we should also address providing context to the individual `autocomplete-item`s when in focus. Created https://github.com/Esri/calcite-design-system/issues/12505 to accommodate.

**Comment by isaacbraun** [comment]
> Yes, this is expected, so assistive technology would be provided with the results that are visually provided to users. We could work to finesse this a bit in the future where we could provide the number of results first, but it seems like would be an enhancement request or something we could wait for someone to file. WDYT, [@isaacbraun](https://github.com/isaacbraun)?

Not sure if I explained this clearly. For example, if you type in "Red", it will read out all the options fetched for just the first character, "R", then all the options for "Re", and finally all the options for "Red." Again, this is likely the native behavior of `aria-live`, but to me it was confusing. Additionally, I agree that providing the number of results would be helpful.

I found [these examples](https://alphagov.github.io/accessible-autocomplete/examples/) from the [UK Gov design systems' component](https://github.com/alphagov/accessible-autocomplete) that IMO provides a greatly improved UX for screen readers. It reads out the number of list items available with each keystroke, and only reads the options as you cycle through them. 

> There are some UX considerations around implementing this should be considered. It would likely be a poor UX to read aloud every item in the list box for a search term, so an alternative approach may be preferable for SR users such as reading aloud the number of results.

Chris spoke about this in the initial issue submission. I think this is something we should seriously consider implementing, so my question now is should we install the current change as a stop-gap and create an enhancement issue for a more complex solution? @geospatialem @DitwanP 

Thanks for the help and for creating #12505 👍

**Comment by isaacbraun** [comment]
Verified on `3.3.0-next.36` via https://codepen.io/isaacbraun/pen/pvjvbvX. Follow up enhancements may happen with #12515 and #12505.
